vcr/matchers.py

```diff
@@ -30,12 +32,29 @@ def query(r1, r2):
     return r1.query == r2.query
 
 
-def body(r1, r2):
+def raw_body(r1, r2):
     if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
         return r1.body.read() == r2.body.read()
     return r1.body == r2.body
 
 
+def body(r1, r2):
+    if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
+        r1_body = r1.body.read()
+        r2_body  = r2.body.read()
+    else:
+        r1_body = r1.body
+        r2_body  = r2.body
+    if r1.headers.get('Content-Type') == r2.headers.get('Content-Type') == 'application/x-www-form-urlencoded':
```

Comment:
Yup, in that case it's fine, as long as it's just used for this matcher.  Maybe we could add it to `vcr/datastructs.py`?

I can picture someone wanting to write a custom matcher where they care about header capitalization, so I don't think I want to change the matcher API - just use the CaseInsensitiveDict for looking up the content-type of our "smart body matcher".

This raises questions about whether our header matcher should also do some kind of case-insensitive match, but I can see both sides on that one.